Sources/GRPCInterceptors/Tracing/SpanAttributes+GRPCTracingKeys.swift

```diff
@@ -0,0 +1,132 @@
+    self.attributes[GRPCTracingKeys.networkTransport] = networkTransportMethod
+    self.attributes[GRPCTracingKeys.rpcService] = context.descriptor.service.fullyQualifiedService
+    self.attributes[GRPCTracingKeys.rpcMethod] = context.descriptor.method
+
+    // Set server address information
+    switch PeerAddress(context.remotePeer) {
+    case .ipv4(let address, let port):
+      self.attributes[GRPCTracingKeys.networkType] = "ipv4"
+      self.attributes[GRPCTracingKeys.networkPeerAddress] = address
+      self.attributes[GRPCTracingKeys.networkPeerPort] = port
+      self.attributes[GRPCTracingKeys.serverPort] = port
+
+    case .ipv6(let address, let port):
+      self.attributes[GRPCTracingKeys.networkType] = "ipv6"
+      self.attributes[GRPCTracingKeys.networkPeerAddress] = address
+      self.attributes[GRPCTracingKeys.networkPeerPort] = port
+      self.attributes[GRPCTracingKeys.serverPort] = port
+
+    case .unixDomainSocket(let path):
+      self.attributes[GRPCTracingKeys.networkType] = "unix"
+      self.attributes[GRPCTracingKeys.networkPeerAddress] = path
+
+    case .other(let address):
+      // We can't nicely format the span attributes to contain the appropriate information here,
+      // so include the whole thing as part of the peer address.
+      self.attributes[GRPCTracingKeys.networkPeerAddress] = address
+    }
+  }
+}
+
+package enum PeerAddress: Equatable {
+  case ipv4(address: String, port: Int?)
+  case ipv6(address: String, port: Int?)
+  case unixDomainSocket(path: String)
+  case other(String)
+
+  package init(_ address: String) {
+    // We expect this address to be of one of these formats:
+    // - ipv4:<host>:<port> for ipv4 addresses
+    // - ipv6:[<host>]:<port> for ipv6 addresses
+    // - unix:<uds-pathname> for UNIX domain sockets
+
+    // First get the first component so that we know what type of address we're dealing with
+    let addressComponents = address.split(separator: ":", maxSplits: 1)
+
+    guard addressComponents.count > 1 else {
+      // This is some unexpected/unknown format, so we have no way of splitting it up nicely.
+      self = .other(address)
+      return
+    }
+
+    // Check what type the transport is...
+    switch addressComponents[0] {
+    case "ipv4":
+      let ipv4AddressComponents = addressComponents[1].split(separator: ":")
+      if ipv4AddressComponents.count == 2, let port = Int(ipv4AddressComponents[1]) {
+        self = .ipv4(address: String(ipv4AddressComponents[0]), port: port)
+      } else {
+        // This is some unexpected/unknown format, so we have no way of splitting it up nicely.
+        self = .other(address)
+      }
+
+    case "ipv6":
+      // At this point, we are looking at an address with format: [<address>]:<port>
+      // We drop the first character ('[') and split by ']:' to keep two components: the address
+      // and the port.
+      let ipv6AddressComponents = addressComponents[1].dropFirst().split(separator: "]:")
```

Comment:
I think we should be a bit stricter here and check that we actually have a "[". 

We also have plenty of allocations here which are mostly avoidable (which I'm fine to punt on for now), but could you file an issue? The idea would be to avoid `split` and all the extra `String` conversions by using the UTF8 view.

Reply:
https://github.com/grpc/grpc-swift-extras/issues/30